Sources/NautilusTelemetry/Tracing/Span+URLSession.swift

```diff
@@ -290,6 +286,56 @@ extension Span {
 		return Int64(duration * 1_000_000_000)
 	}
 
+	/// Add URLRequest attributes to the span
+	/// - Parameters:
+	///   - request: request to fetch attributes from
+	///   - captureHeaders: a list of headers to capture. If nil, none will be captured
+	///   - urlRedaction: A closure to map an URL into a String, redacting sensitive data. If not provided, a default implementation is used.
+	func addRequestAttributes(_ request: URLRequest, captureHeaders: Set<String>? = nil, urlRedaction: ((URL) -> String?)? = nil) {
+		addAttribute("http.request.method", request.httpMethod ?? "_OTHER")
+		addAttribute("user_agent.original", request.value(forHTTPHeaderField: "user-agent"))
+
+		if let url = request.url {
+			addAttribute("server.address", url.host)
+			addAttribute("server.port", url.port)
+			addAttribute("url.scheme", url.scheme)
+
+			let urlRedaction = urlRedaction ?? defaultUrlRedaction
```

Comment:
I wonder if this is the default behavior we want. It means that there isn't a way to skip redaction without passing in a non-redacting closure.

I'd lean toward _not_ providing a default implementation and doing that in user code. We could provide a reference implementation as an example or as a public function that the caller could explicitly pass.